datadog/utf_test.go

```diff
@@ -0,0 +1,43 @@
+package datadog
+
+import (
+	"bytes"
+	"testing"
+)
+
+var testUnicodeByteArrays = []struct {
+	input    []byte
+	expected []byte
+}{
+	// Valid metric
+	{
+		input:    []byte("metric.name:1|c|#tag1:value1,tag2:value2"),
+		expected: []byte("metric.name:1|c|#tag1:value1,tag2:value2"),
+	},
+	// Metric with invalid UTF-8 at the start
+	{
+		input:    []byte("\xff\xfe.metric:100|h|#tag:value"),
+		expected: []byte("\uFFFD\uFFFD.metric:100|h|#tag:value"),
+	},
+	// Metric with invalid UTF-8 embedded
+	{
+		input:    []byte("metric_name:\xe2\x26|h|#tag1:\xffvalue;tag2:value2"),
+		expected: []byte("metric_name:\uFFFD&|h|#tag1:\uFFFDvalue;tag2:value2"),
+	},
+	// Metric with invalid UTF-8 at the end
+	{
+		input:    []byte("metric.value:123|g|#\xff\xfe"),
+		expected: []byte("metric.value:123|g|#\uFFFD\uFFFD"),
```

Comment:
Let's add a test that valid multi-byte UTF-8 characters - for example, emojis - get passed through correctly.

Reply:
sure. will do it.